src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerBase.java

```diff
@@ -551,6 +551,14 @@ public void resolve(DeserializationContext ctxt) throws JsonMappingException
                 }
                 SettableBeanProperty newProp = prop.withValueDeserializer(deser);
                 _replaceProperty(_beanProperties, creatorProps, prop, newProp);
+            } else if (creatorProps != null) {
+                for (int i = 0, len = creatorProps.length; i < len; ++i) {
```

Comment:
This feels wrong, as in it should not be necessary. But I'll need to spend some time figuring out why creator property in question was not updated.

Looking bit further ahead, commented out code-section at line 687 might be what is needed, and/or indicates there is a problem.

Reply:
Nope, that won't help as apparently property has value deserializer but somehow `creatorProps` has not been properly updated earlier.

Reply:
Ok, I guess `creatorProps` returned by `[Std]ValueInstantiator` has not been synced with changes made via `updateBuilder()`. So that is legit problem to resolve, just not sure where it should be done.

Reply:
Now that #4515 is completed, can rule out problem with linkage. I do think that `BeanDeserializerBuilder` needs access to the property-based creator, and that the issue here really is that `CreatorProperty` within that Creator does not properly updated when property in `_properties` is updated.

So the question is then how to pass it along, update properly.